vector3d pos
string name
